src/vm/eetwain.cpp

```diff
@@ -3099,7 +3099,17 @@ size_t EECodeManager::GetCallerSp( PREGDISPLAY  pRD )
     {
         EnsureCallerContextIsValid(pRD, NULL);
     }
-    return (size_t) (GetSP(pRD->pCallerContext));
+
+    size_t res = GetSP(pRD->pCallerContext);
+
+#ifdef _TARGET_X86_
+    EECodeInfo codeInfo;
+    codeInfo.Init((PCODE) pRD->ControlPC);
+
+    res -= codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
```

Comment:
@parjong I am confused by your reply. My comment was not suggesting a change in the behavior, but rather "caching" the value we assign to EstablisherFrame in the OOPStackUnwinderX86::VirtualUnwind, since that's exactly the value we would compute here. So the point was that we don't have to recompute it here that way.